src/main/java/nextstep/subway/domain/Sections.java

```diff
@@ -0,0 +1,197 @@
+package nextstep.subway.domain;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import javax.persistence.CascadeType;
+import javax.persistence.Embeddable;
+import javax.persistence.OneToMany;
+
+@Embeddable
+public class Sections {
+
+    @OneToMany(
+            mappedBy = "line",
+            cascade = {CascadeType.PERSIST, CascadeType.MERGE},
+            orphanRemoval = true)
+    private List<Section> sections = new ArrayList<>();
+
+    public void initSection(Section newSection) {
+        sections.add(newSection);
+    }
+
+    public void addSection(Section newSection) {
+        if(sections.isEmpty()){
+            sections.add(newSection);
+            return;
+        }
+
+        validateAddSectionRule(newSection);
+
+        List<Station> sortedStations = getSortedStations();
+        Station lastUpStation = sortedStations.get(0);
+        Station lastDownStation = sortedStations.get(sortedStations.size()-1);
+
+        if(canAddSectionToUpOrDown(newSection, lastUpStation, lastDownStation)){
+            sections.add(newSection);
+            return;
+        }
+
+        addSectionToMiddle(newSection);
+    }
+
+    private void addSectionToMiddle(Section newSection) {
+        Section targetSection = sections.stream()
+                .filter(e -> e.getUpStation().getId() == newSection.getUpStation().getId()
+                        || e.getDownStation().getId() == newSection.getDownStation().getId())
+                                        .findFirst()
+                                        .orElseThrow(IllegalArgumentException::new);
+
+        if (targetSection.getDistance() <= newSection.getDistance()) {
+            throw new IllegalArgumentException("추가된 구간이 기존 구간보다 길이가 깁니다.");
+        }
+
+        sections.add(newSection);
+
+        if(targetSection.getUpStation().getId() == newSection.getUpStation().getId()){
```

Comment:
```suggestion
        if (targetSection.equalsUpStation(newSection)) {
```
getter로 값을 꺼내서 비교하지 말고 객체에게 물어보는게 어떨까요?
Sections가 아닌 Section 객체를 직접 쓸때도 동일한 로직을 쓸 수 있습니다.
저 부분에 대한 테스트 코드만 따로 작성할 수 있는 장점도 있겠네요.